### DISCUSSION TITLE
Enabling the use of otvdm on Windows ARM?

### DISCUSSION THREAD

**Discussion by LLindfors61** [otya128/winevdm#1041] [Ideas]
Hi there!

I am a scientist and have been using a program called SigmaPlot since the 90's. Although there are newer versions out there I am very fond of SigmaPlot version 2.01 - which is a 16bit application. In my work I am using an iMac with a Parallels Desktop virtualized Windows. For different reasons, I was recently forced to replace my old iMac (2013) with a brand new M1 iMac - a terrific piece of computer, but unfortunately not equipped with an Intel processor. So, to continue to use Parallels Desktop I have to use Windows 11 ARM - with no support of ntvdm (previously available in x86). I have tested your phenomenal otvdm on a x64 Windows 10 remote PC (otvdm-master-2141) - works like a charm. Now to the super-duper naive question - is it at all possible to get otvdm working on Windows 11 ARM?

Looking forward to your thoughts (mission impossible or?)

Lennart

**Comment by Nothoro** [comment]
Hi, I just thought I would kick my two cents in here. I am using Windows 11 ARM x64 build 22000 release 210604 on a raspberry pi400. I don't seem to have any problems running otvdm with anything I have tested with yet, usually things I write in Borland Delphi and Visual Basic 3. The only problem I have is that it wont properly import registry entries for a proper installation so that I would be able to just click and run things. I need to run otdvdmw.exe and browse/locate my files to get things going, which is ok with me. I haven't tested it with SigmaPlot v2.01 as I dont have it.
Best Regards,
Brian.

**Reply by Nothoro** [reply]
Just a side note, if I need to (click n run) something I just create a shortcut to the program to otdvdm.exe and in the target, just add the name of your desired program to launch.

**Reply by LLindfors61** [reply]
Hi!

And thanks for the note! As I mentioned, using otvdm on x64 remote PC SigmaPlot works as normal (so far not a single problem noted). With an identical approach but on my virtual Windows 11 ARM (on my M1 iMac), I can open the program but as soon as try e.g. to open a SigmaPlot file or plot some dat, I get the spinning wheel. Would it be possible for you to share one functional program so I could test on my virtual Windows 11 ARM?

Best regards
Lennart

P.S. I do exactly the same with the shortcut (works on x64 but not on ARM)

**Reply by Nothoro** [reply]
When you say share a functional program, I can direct you to something that Ive written or you can grab something from archive.org:

https://archive.org/details/softwarelibrary_win3

If you want me to compile something real quick, I can do that.

**Reply by LLindfors61** [reply]
Terrific - you can just point to any of your programs that you have found to work on Windows 11 ARM. Now I have to ask a somewhat “stupid” question - when I said I got things to work on x64 I have just downloaded the otvdm executable (otvdm-master-2141) and that is also what I have tried on Windows 11 ARM - thus, I have not compiled anything (either on x64 or ARM) - is this what you do as well?

**Comment by emendelson** [comment]
I've been trying otvdm on Windows 11 ARM on an M1 Mac, via Parallels. I can't get any application to start, so I'm impressed that SigmaPlot will at least open...

**Reply by LLindfors61** [reply]
Hi!

Have you tried November-master-2141? Because with “lower numbers” I had the same experience.

Regards,

Lennart

**Reply by LLindfors61** [reply]
Spell correcting I mean otvdm-master-2141

**Comment by otya128** [comment]
Fixed an issue in 8769a4a2f4f84ef3412fd7ea5ce98df86222e095 that didn't work with aarch64 + mac + Parallels + Windows.
Only in M1 mac environment, there was a problem that the CS and DS values returned by mov selector, ax and GetThreadContext in vm86.dll were random.

**Comment by DougJoseph** [comment]
I am trying to install [otvdm-v0.8.1.zip](https://github.com/otya128/winevdm/releases/download/v0.8.0/otvdm-v0.8.1.zip) on a VM of Windows 11 Pro (ARM edition) running on a host MacOS computer that has M1 Max chip. 

The "install" link provided in the download does not result in an install. I get the following error in a full halt:

<img width="371" alt="Screenshot 2023-09-08 at 7 15 42 PM" src="https://github.com/otya128/winevdm/assets/8921397/9a29ed9f-2a66-46cb-8392-7633112f60f2">

I'm new to otvdm. Is there any chance I can manually install it by copying/moving some files?

**Comment by DougJoseph** [comment]
Also, when I manually copy the [otvdm-v0.8.1.](https://github.com/otya128/winevdm/releases/download/v0.8.0/otvdm-v0.8.1.zip) files to the C drive in the VM of Windows 11 Pro (ARM-64) and double click on otvdmw.exe and then browse to an old program (originally designed for Windows 3.1), I get the following error.

<img width="428" alt="Screenshot 2023-09-08 at 8 02 53 PM" src="https://github.com/otya128/winevdm/assets/8921397/a67325b6-65a8-4f42-a92b-bd749b12893a">

**Reply by otya128** [reply]
Please try the latest version.

[otvdm-master-2446.zip](https://github.com/otya128/winevdm/files/12565710/otvdm-master-2446.zip)


If the error occurs even with the latest version, please create an issue with the program information.

**Comment by DougJoseph** [comment]
> Please try the latest version.

You are amazing. I will try and report back.

**Comment by DougJoseph** [comment]
OK, reporting back on the test of otvdm-master-2446 on Windows 11 Pro ARM-64 editing running inside a UTM Virtual Machine running on a MacOS host on MacBookPro running an M1 Max chip:

I have awesome news, and an install fail note! 

First, in the way of awesome news, I manually copied the extracted folder contents to the C drive, ran the otvdmw.exe file, and then browsed to a software program originally made for Windows 3.1 (in this case, QuickVerse Bible program), and it opened and ran flawlessly!

<img width="1728" alt="Screenshot 2023-09-10 at 12 22 52 AM" src="https://github.com/otya128/winevdm/assets/8921397/2b7dcd72-d524-4f42-83e1-2826f550f71d">

And even copying from the old program and pasting in the modern Windows 11 side works!

<img width="1728" alt="Screenshot 2023-09-10 at 12 28 34 AM" src="https://github.com/otya128/winevdm/assets/8921397/5230b744-ed22-4588-9b9c-096af3f25cc3">

I am over the moon excited about this! Thank you!!

I do need to also mention that the install link is still saying this while failing to install:

<img width="363" alt="Screenshot 2023-09-09 at 11 56 27 PM" src="https://github.com/otya128/winevdm/assets/8921397/5ad23d14-1d40-40fe-9830-2851def260c9">

Thanks again. This is amazing.
